.github/workflows/build-and-test.yml

```diff
@@ -8,7 +8,33 @@ on:
   pull_request:
     branches: [ main ]
 jobs:
+  detect-changes:
+    runs-on: ubuntu-latest
+    outputs:
+      modules: ${{ steps.find-modules.outputs.changed_modules }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Identify Changed Modules
+        id: find-modules
+        run: |
+          # Alle Maven-Module (Anhand der POM-Dateien suchen)
+               ALL_MODULES=$(git ls-tree -d HEAD | awk '{print $4}' | jq -R -s -c 'split("\n") | map(select(length > 0))')
+
+          # Liste der Module filtern, die sich geändert haben
+               CHANGED_MODULES=$(git diff --name-only HEAD^ HEAD | awk -F'/' '{print $1}' | sort -u | jq -R -s -c 'split("\n") | map(select(length > 0))')
```

Comment:
`HEAD^` is equivalent to `HEAD^1`. This only detects changes from the last commit but a branch or PR may contain multiple commits across several modules.

You would need to compare this to the target branch of a PR or the main branch. Maybe we can get this in a dedicated PR as it is not really related to the release workflow.